README.md

```diff
@@ -1,8 +1,29 @@
 # Task
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/task`. To experiment with that code, run `bin/console` for an interactive prompt.
+Task provides a toolbox for generating, tracking and serializing tasks to be performed.
+It is NOT a job queuing framework, a la resque or qless, and does not provide ways to execute tasks
+in a distributed fashion. Rather, Task could be used with such a framework to provide "at-least-once"
+task execution guarantees, or to perform multiple, lightweight tasks within a single job.
 
-TODO: Delete this and the text above, and describe your gem
+An example task lifecycle could be follows:
+
+1. A task is generated
+2. The task is saved to a backing data store, recording that it has been generated and should be completed.
+3. The task is serialized (as JSON) and passed to another process or host to complete.
+4. The worker process begins working on the task, and:
+  * The worker process completes the task, marking it as completed in the backing data store.
+
+  OR
+
+  * The worker process fails to complete the task.
+  * A subsequent worker can fetch the task from the backing data store and attempts to complete it.
+
+Out of the box, Task uses Cassandra as the backing data store, but other backends could be implemented.
```

Comment:
maybe provide some insight as to what a datastore needs to provide in order to be valid / useful to this task model.  for example, is it the fact the cassandra and / or redis is able to provide a key / value store?  does it make the most sense to choose a data store which performs this operation very well?  this might be good insight for someone looking to swap the datastore and is either not as familiar with the options listed here or that might not be feasible for some other reason.